src/math/TransformBoundedField.cc

```diff
@@ -0,0 +1,172 @@
+
+    // res2D has shape 1 x N; return a 1-D view with the extra dimension stripped
+    auto resShape = ndarray::makeVector(nPoints);
+    auto resStrides = ndarray::makeVector(1);
+    return ndarray::external(res2D.getData(), resShape, resStrides, res2D);
+}
+
+// ------------------ persistence ---------------------------------------------------------------------------
+
+namespace {
+
+struct PersistenceHelper {
+    table::Schema schema;
+    table::Key<int> orderX;
+    table::PointKey<int> bboxMin;
+    table::PointKey<int> bboxMax;
+    table::Key<std::string> frameSet;
+
+    PersistenceHelper()
+            : schema(),
+              bboxMin(table::PointKey<int>::addFields(schema, "bbox_min", "lower-left corner of bounding box",
+                                                      "pixel")),
+              bboxMax(table::PointKey<int>::addFields(schema, "bbox_max",
+                                                      "upper-right corner of bounding box", "pixel")),
+              frameSet(schema.addField<std::string>("frameSet", "FrameSet contained in the Transform", 0,
+                                                    false)) {}
+
+    PersistenceHelper(table::Schema const& s)
+            : schema(s), bboxMin(s["bbox_min"]), bboxMax(s["bbox_max"]), frameSet(s["frameSet"]) {}
+};
+
+class TransformBoundedFieldFactory : public table::io::PersistableFactory {
+public:
+    virtual std::shared_ptr<table::io::Persistable> read(InputArchive const& archive,
+                                                         CatalogVector const& catalogs) const {
+        LSST_ARCHIVE_ASSERT(catalogs.size() == 1u);
+        LSST_ARCHIVE_ASSERT(catalogs.front().size() == 1u);
+        table::BaseRecord const& record = catalogs.front().front();
+        PersistenceHelper const keys(record.getSchema());
+        // NOTE: needed invert=false in case min=-1, max=0 (empty bbox). See RFC-324 and DM-10200
+        geom::Box2I bbox(record.get(keys.bboxMin), record.get(keys.bboxMax), false);
+        auto frameSetStr = record.get(keys.frameSet);
+        auto stream = ast::StringStream(frameSetStr);
+        auto astObjectPtr = ast::Channel(stream).read();
+        auto frameSet = std::dynamic_pointer_cast<ast::FrameSet>(astObjectPtr);
+        if (!frameSet) {
+            throw LSST_EXCEPT(pex::exceptions::RuntimeError,
+                              "The contents of the frameSet field could not be parsed as an ast::FrameSet");
+        }
+        auto transform = geom::Transform<geom::Point2Endpoint, geom::GenericEndpoint>(*frameSet);
+        return std::make_shared<TransformBoundedField>(bbox, transform);
+    }
+
+    TransformBoundedFieldFactory(std::string const& name) : afw::table::io::PersistableFactory(name) {}
+};
+
+std::string getTransformBoundedFieldPersistenceName() { return "TransformBoundedField"; }
+
+TransformBoundedFieldFactory registration(getTransformBoundedFieldPersistenceName());
```

Comment:
I am very reluctant to depart from the way ChebyshevBoundedField is doing persistence without understanding it a lot better.